### ISSUE TITLE
Impossible to connect Azure ML Compute Instance via VS code

### ISSUE THREAD

**Issue by pr-scandas** [microsoft/vscode-tools-for-ai#2025]
Does this occur consistently? Yes
Repro steps:

Trial 1: From Azure extension, connect to Machine Learning Compute Instance
Trial 2: From Azure ML workspace, Select editor as VS Code

Action: Resolver.resolve
Error type: 70
Error Message: request to redacted:url failed, reason: unable to get local issuer certificate

Version: 0.28.1
OS: win32
OS Release: 10.0.22621
Product: Visual Studio Code
Product Version: 1.78.0
Language: en

<details>
<summary>Call Stack</summary>

```
s extension.js:2:1981706
extension.js:2:2007388extension.js:2:2007388
```

</details>

**Comment by sevillal** [comment]
Hi @pr-scandas , I apologize for the issue you are facing.

I would like to get more details about your scenario:
- Is your workspace secured with Private Link and you compute instance is inside a Vnet?
- Is the compute instance failing to connect recently created?
- Were you able to connect to the failing compute instance before?
- Were you able to connect to other compute instances before?
- Can you try connecting to that compute instance from a different machine? (just in case you have access to another computer)

**Comment by sevillal** [comment]
Thanks @janmolemans for sharing that workaround!

**Comment by sevillal** [comment]
@pr-scandas and @janmolemans could you please try installing this extension to diagnose problems like this: https://marketplace.visualstudio.com/items?itemName=chrmarti.network-proxy-test. Then use F1 > Network Proxy Test: Test Network Connection with the URL is failing (I assume it might be something like https://<COMPUTE-INSTANCE>.<REGION>.instances.azureml.ms/api/terminals) to check if that works. If it shows the certificate error retry with F1 > Network Proxy Test: Test Network Connection (Allow Unauthorized) to get more information about the problem.

Make sure replace any sensitive information before posting it here. Thanks.

**Comment by pr-scandas** [comment]
Hello @sevillal 
Here are my answers : 
- Is your workspace secured with Private Link and you compute instance is inside a Vnet? --> Yes
- Is the compute instance failing to connect recently created? --> No, everything was working correctly before Friday 5th May
- Were you able to connect to the failing compute instance before? --> Yes
- Were you able to connect to other compute instances before? --> Yes
Can you try connecting to that compute instance from a different machine? (just in case you have access to another computer) --> I tried from my personal laptop (not the one used by my company)

**Comment by pr-scandas** [comment]
> Downgrading to vscode 1.77.3 solved the issue for me.

Nice to read, it didn't worked for me

**Comment by pr-scandas** [comment]
Hello @sevillal 
Here is what I get from the Network Proxy Test

`Note: Make sure to replace all sensitive information with dummy values before sharing this output.

VS Code 1.78.0 (252e5463d60e635E345799aef7375787f68b4ee)
Network Proxy Test 0.0.6
win32 10.0.22621 x64

Settings:
- http.proxy: 
- http.proxyAuthorization: null
- http.proxyStrictSSL: true
- http.proxySupport: override
- http.systemCertificates: true

Environment variables:

Sending GET request to https://************.instances.azureml.ms/api/terminals...
vscode-proxy-agent: DIRECT
Received error: unable to get local issuer certificate (UNABLE_TO_GET_ISSUER_CERT_LOCALLY)
Retrying while ignoring certificate issues to collect information on the certificate chain.

Sending GET request to https://************.instances.azureml.ms/api/terminals (allowing unauthorized)...
vscode-proxy-agent: DIRECT
Received response:
- Status: 302 Found
- Location: https://login.microsoftonline.com/common/oauth2/authorize?client_id=2345324532452345252345&redirect_uri=https%3A%2F%2Fauth.*.instances.azureml.ms%2Fsignin-oidc3&response_type=id_token&scope=openid%20profile%20offline_access&response_mode=form_post&nonce=638192180617675059GFDSSDFGZEDFGZER&client_info=1&state=CfDJ8F-A9eo3zr9CvpUsPiPF5gqXAudqrIG-WMXagcSYkqIedKJnbHfSbDELA4nip-RDR-LXOaeO8x01RyB_yHF49iZ9jLdzD3MtY60eHtBeRN4t8baTitnke2ZQiC0vLFnaMUE11GhxPyOkJd47odZxzE9sZ_sSwTmmlmUvhzAJMMC6ofUA3exq2vWtY-6jhLqF6Cdo-nm2gXkzhp_roTam-qsjlkdfhqslkdjfhlgqsldkfgql-E7i6wrUQBvyG3zPqIY7w7m78QHLd2aUjhBh9ULkqwQ&x-client-SKU=ID_NETSTANDARD2_0&x-client-ver=6.10.0.0
Certificate chain:
- Subject: *.*.instances.azureml.ms (Microsoft Corporation)
  Subject alt: DNS:*.*.instances.azureml.ms, DNS:*.instances.azureml.ms
  Validity: May  6 04:54:35 2023 GMT - May 20 04:54:35 2023 GMT
  Fingerprint: CB:E1:4D:7C:A3:**:**:**:**:**:**:06:D6:33:60:6A:DC:97:CE
- Subject: digo-digo.com (t)  (digo-digo)
  Validity: May  6 04:54:35 2023 GMT - May 20 04:54:35 2023 GMT
  Fingerprint: 69:B5:A7:02:1E:C4:**:**:**:**:**:88:CE:33:64:8A:82:5B:17
- Subject: digo-digo.com (digo-digo)
  Validity: Sep 17 06:55:53 2020 GMT - Sep 17 07:05:53 2023 GMT
  Fingerprint: 5A:43:DD:99:00:37:D5:99:4F:**:**:**:**:**:A4:39:B6:48:50

Issuer certificate '****-CA' not found. This might indicate an issue with the root certificates registered in your OS:
- Make sure that the root certificate for the certificate chain is registered as such in the OS. Use `F1` > `Network Proxy Test: Show OS Certificates` to see the list loaded by VS Code.
- Also make sure that your proxy and server return the complete certificate chain (except for the root certificate).
`

**Comment by pr-scandas** [comment]
Hi @sevillal 
Actually, I already executed this command and the certificate mentioned above is already in the list of certificated loaded by VSCode. 

Is there anything else to check ?

**Comment by sevillal** [comment]
@pr-scandas could you please [try ](https://github.com/microsoft/vscode-tools-for-ai/issues/2037#issuecomment-1542774582)this and let me know if that works.

**Comment by pr-scandas** [comment]
Hello @sevillal, happy to share that downgrading the version resolved the issue on my side also. In my first attempt, I was trying to downgrade my VSCode version.

**Comment by sevillal** [comment]
@pr-scandas @janmolemans VS Code just released version 1.78.2 which should fix the certificates issue, could you please update VS Code (Help > Check for Updates...) and retry the connection.
